src/components/TokenHolderList/TokenHolderList.tsx

```diff
@@ -0,0 +1,80 @@
+import { useMemo } from 'react';
+import styled from 'styled-components';
+import { TitleS } from 'components/core/Text/Text';
+import Spacer from 'components/core/Spacer/Spacer';
+import TokenHolderListItem from './TokenHolderListItem';
+import { Directions } from 'src/components/core/enums';
+import { removeNullValues } from 'utils/removeNullValues';
+import { useMemberListPageState } from 'contexts/memberListPage/MemberListPageContext';
+import { MemberListTierFragment$data } from '__generated__/MemberListTierFragment.graphql';
+import { CommunityPageViewFragment$data } from '__generated__/CommunityPageViewFragment.graphql';
+
+// Get which side of the tokenHolder name to show the preview on
+// 1st and 2nd column should be right, 3rd and 4th column should be left
+function getPreviewDirection(index: number) {
+  return index % 4 < 2 ? Directions.RIGHT : Directions.LEFT;
+}
+
+type Props = {
+  title: string;
+  tokenHoldersRef: MemberListTierFragment$data['owners'] | CommunityPageViewFragment$data['owners'];
```

Comment:
This component is used by both the Community Page and Membership Directory Page, which have query fragments on different types: `Community` and `MembershipTier`. Both of these types have an `owners` field, which is a list of `TokenHolder`s. 

Passing in this common field, `owners` rather than the parent fragment satisfied the type requirements and allowed me to share the `TokenHolderListItemFragment` child fragment for both the Membership Directory and Community queries.

Reply:
sounds good to me! @tbezman is this generally best practice?